src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +86,15 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> filteredPerson = p -> p.getSex() == Person.Sex.WOMAN
+                && p.getAge() >= femaleAge
+                && p.getCats().size() > 0;
+        return peopleList.stream()
+                .filter(filteredPerson)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
```suggestion
                .flatMap(w -> w.getCats().stream())
```
Better to use as less **map**s as possible